addons/mail/models/mail_thread.py

```diff
@@ -1134,6 +1135,18 @@ def message_route(self, message, message_dict, model=None, thread_id=None, custo
         if rcpt_tos_localparts:
             # no route found for a matching reference (or reply), so parent is invalid
             message_dict.pop('parent_id', None)
+
+            # check it does not directly contact catchall
+            if catchall_alias and catchall_alias in email_to_localpart:
+                _logger.info('Routing mail from %s to %s with Message-Id %s: direct write to catchall, bounce', email_from, email_to, message_id)
+                body = """<div>
+<p>Hello,</p>
+<p>The following email sent to %s cannot be accepted because this is a catchall email
+   used to collect replies. BLAH BLAH BLAH</p>
```

Comment:
blah blah blah ?

Reply:
Yeah. I wonder if boom boom boom boom boom if not better.